shared/src/test/diff/codegen/NewMatching.mls

```diff
@@ -264,13 +250,17 @@ fun c(x) =
   if x is
     VVC(x, y, z) then x + y + z
     _ then 0
-//│ ╔══[ERROR] class VVC expects 2 parameters but found 3 parameters
-//│ ║  l.265: 	    VVC(x, y, z) then x + y + z
-//│ ╙──       	    ^^^^^^^^^^^^
+//│ ╔══[ERROR] Type mismatch in field selection:
+//│ ╟── tuple literal of type `{0: ?#v, 1: ?#vc}` does not have field '2'
+//│ ║  l.248: 	class VVC(v: Int, vc: Int)
+//│ ║         	          ^^^^^^^^^^
+//│ ╟── but it flows into operator application with expected type `{2: ?a}`
+//│ ║  l.250: 	  if x is
+//│ ║         	     ^^^^
+//│ ║  l.251: 	    VVC(x, y, z) then x + y + z
+//│ ╙──       	^^^^^^^
```

Comment:
The old desugarer generates better messages. We should polish the error message.